core/src/labels/spriteLabel.cpp

```diff
@@ -12,28 +12,16 @@ void SpriteLabel::pushTransform(VboMesh& _mesh) {
     if (m_dirty) {
         TypedMesh<BufferVert>& mesh = static_cast<TypedMesh<BufferVert>&>(_mesh);
 
-        // used to write all attributes memory at once, good for caching
-        struct VertexAttributes {
-            glm::vec2 screenPos;
-            float alpha;
-            float rot;
-        };
-
-        GLintptr stride = std::min<GLintptr>(m_attribOffsets.m_position, std::min<GLintptr>(m_attribOffsets.m_alpha, m_attribOffsets.m_rotation));
-
-        VertexAttributes newAttributes {
-            m_transform.m_screenPosition + m_offset,
-            m_transform.m_alpha,
-            m_transform.m_rotation,
-        };
+        BufferVert::State state(m_transform.state);
+        state.screenPos += m_offset;
```

Comment:
I think we could skip this extra copy by moving `m_offset` up to the label class, (so that text label would also benefit from it) and apply it in `updateScreenTransform`.